### PR TITLE
[MU4] Audio output

### PR DESCRIPTION
**Pull request by Eism**

### CODE REVIEW

**Review by RomanPudashkin — APPROVED** [review]

File: src/framework/audio/internal/audiooutputcontroller.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        audioDriver()->selectOutputDevice(preferredDeviceId);
+        return;
+    }
+
+    if (!containsDevice(devices, currentDeviceId)) {
+        audioDriver()->resetToDefaultOutputDevice();
+    }
+}
+
+void AudioOutputController::connectCurrentOutputDevice()
```
Comment (cbjeukendrup) [inline src/framework/audio/internal/audiooutputcontroller.cpp:71]: It looks like this method is not called, is it?

**Comment by DmitryArefiev** [comment]
Tested on Win10, Mac12, LinuxMint 20.2 - **now the ability to change audio output device is added** - **let's promote these changes**. Other related issues I'll log separately

Also #11748 is not implemented in full. Let's review it later after buffer size settings will be added.

**Comment by Jojo-Schmitz** [comment]
This breaks the MinWG build:
```
...\src\framework\audio\internal\platform\win\wincoreaudiodriver.cpp:31: Fehler: atlstr.h: No such file or directory
In file included from ...\build.qtc\src\framework\audio\CMakeFiles\audio.dir\Unity\unity_0_cxx.cxx:25:
.../src/framework/audio/internal/platform/win/wincoreaudiodriver.cpp:31:10: fatal error: atlstr.h: No such file or directory
 #include <atlstr.h>
          ^~~~~~~~~~
```

**Comment by Jojo-Schmitz** [comment]
Commenting that line out makes it compile. on MinGW and on MSVC. On MinGW with a copiler warning:
```
...\src\framework\audio\internal\platform\win\wincoreaudiodriver.cpp:55: Warnung: Returning address of local temporary object
```

**Comment by Jojo-Schmitz** [comment]
The warning is indeed in MinGW onky

**Comment by Eism** [comment]
@Jojo-Schmitz  thanks!
Updated PR

**Pull request merged**